src/Receiver.cc

```diff
@@ -212,13 +236,13 @@ Receiver::handlePingPacket(Driver::Packet* packet, IpAddress sourceIp)
     Protocol::MessageId id = header->common.messageId;
 
     MessageBucket* bucket = messageBuckets.getBucket(id);
-    SpinLock::Lock lock_bucket(bucket->mutex);
-    Message* message = bucket->findMessage(id, lock_bucket);
+    Message* message;
+    {
+        SpinLock::Lock lock_bucket(bucket->mutex);
+        message = bucket->findMessage(id, lock_bucket);
+    }
```

Comment:
Message might be invalid after bucket mutex release.

Reply:
Extended the critical section to the entire method.